docs/ctr-remote.md

```diff
@@ -0,0 +1,264 @@
+
+Here, `registry2:5000` in the demo environment serves images with HTTP (not HTTPS) so we need to tell it to the `ctr-remote` command using `--plain-http` option and `http://` prefix.
+
+```
+ctr-remote image optimize --plain-http \
+           ghcr.io/stargz-containers/golang:1.15.3-buster-org \
+           http://registry2:5000/golang:1.15.3-esgz
+```
+
+When you run this command, `ctr-remote` runs the source image (`ghcr.io/stargz-containers/golang:1.15.3-buster-org`) as a container and profiles all file accesses during the execution.
+Then these accessed files are marked as "prioritized" files and will be prefetched on runtime.
+
+You can lazy-pull this image with Stargz Snapshotter.
+The following example lazily pulls this image to containerd, using `ctr-remote image rpull` command (`http://` prefix isn't needed here).
+
+```console
+# ctr-remote image rpull --plain-http registry2:5000/golang:1.15.3-esgz
+egistry2:5000/golang:1.15.3-esgz
+fetching sha256:bd2e5983... application/vnd.oci.image.index.v1+json
+fetching sha256:55aef317... application/vnd.docker.distribution.manifest.v2+json
+fetching sha256:d1123476... application/vnd.docker.container.image.v1+json
+# ctr-remote run --rm -t --snapshotter=stargz registry2:5000/golang:1.15.3-esgz test echo hello
+hello
+```
+
+## Optimizing an image with custom configuration
+
+You can also specify the custom workload configuration that the image is optimized against.
+The following example optimizes the image against the workload running `go version` on `/bin/bash`.
+
+```
+ctr-remote image optimize --plain-http \
+           --entrypoint='[ "/bin/bash", "-c" ]' --args='[ "go version" ]' \
+           ghcr.io/stargz-containers/golang:1.15.3-buster-org \
+           http://registry2:5000/golang:1.15.3-esgz-go-version
+```
+
+Other options are also available for configuring the workload.
+
+|Option|Description|
+---|---
+|`--entrypoint`|Entrypoint of the container (in JSON array)|
+|`--args`|Arguments for the entrypoint (in JSON array)|
+|`--env`|Environment variables in the container|
+|`--user`|User name to run the process in the container|
+|`--cwd`|Working directory|
+|`--period`|The time seconds during profiling the file accesses|
+|`-t`or`--terminal`|Enable to interactively control the container|
+
+## Mounting files from the host
+
+There are several cases where sharing files from host to the container during optimization is useful.
+This includes when we want to optimize an image against building a binary using compilers.
+
+For these use-cases, you can mount files on the hosts to the container using `--mount` option.
+The following example optimizes the image against the workload where Go compiler compiles "hello world" program.
+
+First, create the following Go source file at `/etc/hello.go`.
```

Comment:
Having a Go file under `/etc` looks weird

Reply:
I guess you meant /tmp/hello.go here

Reply:
Exactly.